site/v5/features/features-multitenant.md

```diff
@@ -0,0 +1,114 @@
+toc: true
+eleventyNavigation:
+  version: v5
+  order: 15
+  parent: features
+  key: features multitenant 
+  title: Multitenant
+---
+<h1 class="title">Multitenant <span class="professional"><a href="/pro/index.html">PRO</a></span></h1>
+
+
+
+[[TOC]]
+## Introduction 
+This feature helps in situations when it's required to setup multiple tenants in a single project. A common scenario is a SaaS product in which, instead of having a shared database for all the clients, each has its own and independent database(in the same or different server), but sharing the same source code and deployment. In this case, you would want to apply the same Mongock migration to each client's database. 
+<p class="text-center">
+    <img src="/images/multitenant-overview.png" alt="multitenant overview" width="75%">
+</p> 
+
+## How it works
+The application is configured to work with N datasources and Mongock runs the migration for all of them(sequentially). To achieve this, the user has to configure a Mongock driver per datasource, pass them to the Mongock builder and execute the Mongock runner which will apply the migration for all the tenants.
+
+
+## With Mongock Standalone
+This is the simplest scenario as there is no ORM framework to deal with. It just requires passing a driver per tenant to the Mongock builder. This mechanism is standard and doesn't change among the different Mongock drivers/databases
+```java
+  MongockStandalone.builder()
+    .setDriverMultiTenant(
+      MongoSync4Driver.withDefaultLock(mongoClientTenant1, "db-tenant-1"),
+      MongoSync4Driver.withDefaultLock(mongoClientTenant2, "db-tenant-2")
+    )
+  //...
+```
+
+## With Springdata
+While the core idea and how it works still remains, when using Springdata in a multitenant environment(independently of Mongock) it requires some extra work, which it's explained bellow.
+
+### The problem
+Springdata hides all the complexity to connect to the database, which helps a lot, but it also means the developer needs to give some flexibility up . In this case, he cannot easily specify which tenant is connecting to. 
+
+Lets imagine the following scneario...
+<br />
+
+- We have 3 tenants(one database per each) and and API(`GET /api/{tenant}/products`) that return the list of products per tenant. 
+
+<br />
+
+```java
+@RestController
+public class ClientController  {
+  
+  private final ClientRepository clientRepository;
+
+  public ClientController(ClientRepository clientRepository) {
+    this.clientRepository = clientRepository;
+  }
+  
+  @GetMapping("/api/{tenant}/clients")
+  public List<Client> getClients(@PathVariable("tenant") String tenant) {
+    return clientRepository.findAll();
+  } 
+}
+```
+
+The code above is clearly wrong.
+- We are not using the `tenant` field to indicate the owner of the products. First thinking may be passing it in the `findAll()` method, but remember, each tenant has its own database and the table/collection doesn't have a field `tenant`.
```

Comment:
I'm not fully following what you mean in this phrase

Reply:
with `field`, I meant `parameter`. It basically means that the parameter `tenant` passed to the method in the controller is not used. We're supposed to want to retrieve the products for the given tenant, but the tenant value cannot be passed to the repository's method, as the tenant is not a column in a table. Each tenant has its own database.

Reply:
What about this: 

- The tenant is supposed to be used to retrieve the list of products for a given tenant, but  the above code doesn't use it at all. First thought may be to pass it to the repository's method, like `clientRepository.findAll(tenant)`, but remember that each tenant has its own database, it's not a column in a table which cab be used as query filter